servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/LoadBalancedStreamingHttpClient.java

```diff
@@ -153,8 +155,16 @@ private static void notifyConnectionSelected(final HttpRequestMetaData requestMe
     @Override
     public Single<ReservedStreamingHttpConnection> reserveConnection(final HttpRequestMetaData metaData) {
         return Single.defer(() -> {
-            Single<ReservedStreamingHttpConnection> connection =
-                    loadBalancer.selectConnection(SELECTOR_FOR_RESERVE, metaData.context()).map(identity());
+            final ContextMap context = metaData.context();
+            final Boolean forceNew = context.get(HttpContextKeys.HTTP_FORCE_NEW_CONNECTION);
```

Comment:
Consider using `Boolean.TRUE.equals(context.get(HttpContextKeys.HTTP_FORCE_NEW_CONNECTION))`. This way we can skip autoboxing and use `?:` instead of `if` below